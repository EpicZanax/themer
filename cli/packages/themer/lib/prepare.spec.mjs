import prepareColors from './prepare.mjs';
import { describe, expect, it } from 'vitest';

describe('the color set validation/preparation', () => {
  it('should convert valid CSS colors to hex', () => {
    const result = prepareColors({
      dark: {
        shade0: 'black',
        shade1: 'hsl(0, 0%, 0%)',
        shade2: 'hsla(0, 0%, 0%, 1)',
        shade3: 'rgb(0, 0 ,0)',
        shade4: 'rgb(0%, 0%, 0%)',
        shade5: 'rgba(0, 0, 0, 1)',
        shade6: '#000000',
        shade7: '#000000',
        accent0: '#000000',
        accent1: '#000000',
        accent2: '#000000',
        accent3: '#000000',
        accent4: '#000000',
        accent5: '#000000',
        accent6: '#000000',
        accent7: '#000000',
      },
    });
    expect(result).toMatchSnapshot();
  });
  it('should be idempotent', () => {
    const result1 = prepareColors({
      dark: {
        shade0: '#000',
        shade1: '#111',
        shade2: '#222',
        shade3: '#333',
        shade4: '#444',
        shade5: '#555',
        shade6: '#666',
        shade7: '#777',
        accent0: 'red',
        accent1: 'orange',
        accent2: 'yellow',
        accent3: 'green',
        accent4: 'cyan',
        accent5: 'blue',
        accent6: 'purple',
        accent7: 'pink',
      },
    });
    const result2 = prepareColors(result1);
    expect(result1).toEqual(result2);
  });
  it('should properly fill in shades 1 through 6 when necessary', () => {
    const result = prepareColors({
      dark: {
        shade0: '#000000',
        shade7: '#FFFFFF',
        accent0: 'red',
        accent1: 'orange',
        accent2: 'yellow',
        accent3: 'green',
        accent4: 'cyan',
        accent5: 'blue',
        accent6: 'purple',
        accent7: 'pink',
      },
      light: {
        shade0: '#FFFFFF',
        shade7: '#000000',
        accent0: 'red',
        accent1: 'orange',
        accent2: 'yellow',
        accent3: 'green',
        accent4: 'cyan',
        accent5: 'blue',
        accent6: 'purple',
        accent7: 'pink',
      },
    });
    expect(result).toMatchSnapshot();
  });
  it('should throw when passed an invalid color', () => {
    const fn = () => {
      prepareColors({
        dark: {
          shade0: '#000',
          shade1: '#111',
          shade2: '#222',
          shade3: '#333',
          shade4: '#444',
          shade5: '#555',
          shade6: '#666',
          shade7: '#777',
          accent0: 'red',
          accent1: 'orange',
          accent2: 'yellow',
          accent3: 'green',
          accent4: 'cyan',
          accent5: 'blue',
          accent6: 'purple',
          accent7: 'aif8sd09f8*dasdf7&F^^fajfd',
        },
      });
    };
    expect(fn).toThrow();
  });
  it('should throw if missing a color', () => {
    const fn = () => {
      prepareColors({
        dark: {
          shade0: '#000',
          shade1: '#111',
          shade2: '#222',
          shade3: '#333',
          shade4: '#444',
          shade5: '#555',
          shade6: '#666',
          shade7: '#777',
          accent0: 'red',
          accent1: 'orange',
          accent2: 'yellow',
          accent3: 'green',
          accent4: 'cyan',
          accent5: 'blue',
          accent6: 'purple',
        },
      });
    };
    expect(fn).toThrow();
  });
  it('should throw if missing a shade, even if it is a fillable shade', () => {
    const fn = () => {
      prepareColors({
        dark: {
          shade0: '#000',
          shade1: '#111',
          shade2: '#222',
          shade4: '#444',
          shade5: '#555',
          shade6: '#666',
          shade7: '#777',
          accent0: 'red',
          accent1: 'orange',
          accent2: 'yellow',
          accent3: 'green',
          accent4: 'cyan',
          accent5: 'blue',
          accent6: 'purple',
          accent7: 'pink',
        },
      });
    };
    expect(fn).toThrow();
  });
});
