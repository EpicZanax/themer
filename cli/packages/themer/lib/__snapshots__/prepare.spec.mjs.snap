// Vitest Snapshot v1

exports[`the color set validation/preparation > should convert valid CSS colors to hex 1`] = `
{
  "dark": {
    "accent0": "#000000",
    "accent1": "#000000",
    "accent2": "#000000",
    "accent3": "#000000",
    "accent4": "#000000",
    "accent5": "#000000",
    "accent6": "#000000",
    "accent7": "#000000",
    "shade0": "#000000",
    "shade1": "#000000",
    "shade2": "#000000",
    "shade3": "#000000",
    "shade4": "#000000",
    "shade5": "#000000",
    "shade6": "#000000",
    "shade7": "#000000",
  },
}
`;

exports[`the color set validation/preparation > should properly fill in shades 1 through 6 when necessary 1`] = `
{
  "dark": {
    "accent0": "#FF0000",
    "accent1": "#FFA500",
    "accent2": "#FFFF00",
    "accent3": "#008000",
    "accent4": "#00FFFF",
    "accent5": "#0000FF",
    "accent6": "#800080",
    "accent7": "#FFC0CB",
    "shade0": "#000000",
    "shade1": "#242424",
    "shade2": "#494949",
    "shade3": "#6D6D6D",
    "shade4": "#929292",
    "shade5": "#B6B6B6",
    "shade6": "#DBDBDB",
    "shade7": "#FFFFFF",
  },
  "light": {
    "accent0": "#FF0000",
    "accent1": "#FFA500",
    "accent2": "#FFFF00",
    "accent3": "#008000",
    "accent4": "#00FFFF",
    "accent5": "#0000FF",
    "accent6": "#800080",
    "accent7": "#FFC0CB",
    "shade0": "#FFFFFF",
    "shade1": "#DBDBDB",
    "shade2": "#B6B6B6",
    "shade3": "#929292",
    "shade4": "#6D6D6D",
    "shade5": "#494949",
    "shade6": "#242424",
    "shade7": "#000000",
  },
}
`;
